src/libcore/sync/atomic.rs

```diff
@@ -1550,12 +1550,30 @@ unsafe fn atomic_xor<T>(dst: *mut T, val: T, order: Ordering) -> T {
 
 /// An atomic fence.
 ///
-/// A fence 'A' which has [`Release`] ordering semantics, synchronizes with a
-/// fence 'B' with (at least) [`Acquire`] semantics, if and only if there exists
-/// atomic operations X and Y, both operating on some atomic object 'M' such
+/// A fence creates happens-before relationships between other atomic operations or
+/// fences when the certain condition is met. Also, depending of the specified
+/// `order`, a fence prevents reordering certain types of memory operations
+/// around a fence by CPU or a compiler.
```

Comment:
I agree with you, since rust does not have a signal and `fence` prohibits `Relaxed` ordering.
Thank you very much for your explanation!